src/integrations/front-end/robots-txt-integration.php

```diff
@@ -78,7 +78,17 @@ public function register_hooks() {
 	public function filter_robots( $robots_txt ) {
 		$robots_txt = $this->remove_default_robots( $robots_txt );
 		$this->maybe_add_xml_sitemap();
-		$this->add_subdirectory_multisite_xml_sitemaps();
+
+		/**
+		 * Filter: 'wpseo_ignore_multisite_subdirectory_xml_sitemaps' - Enabling this filter removes subdirectory sites from xml sitemaps.
+		 *
+		 * @since 19.8
+		 *
+		 * @param bool $show Whether to display multisites in the xml sitemaps.
+		 */
+		if ( ! \apply_filters( 'wpseo_ignore_multisite_subdirectory_xml_sitemaps', false ) ) {
```

Comment:
Since this is a boolean value, I suggest to use the name of the method it enables/disables with the `should` prefix:

`wpseo_should_add_subdirectory_multisite_xml_sitemaps `